libs/sql-schema-describer/src/postgres.rs

```diff
@@ -645,49 +691,49 @@ fn get_column_type<'a>(
+        "bit" | "_bit" => (String, Some(PostgresType::Bit(precision.character_maximum_length))),
+        "varbit" | "_varbit" => (String, Some(PostgresType::VarBit(precision.character_maximum_length))),
         "numeric" | "_numeric" => (
             Decimal,
             Some(PostgresType::Numeric(
-                precision.numeric_precision(),
-                precision.numeric_scale(),
+                match (precision.numeric_precision, precision.numeric_scale) {
+                    (None, None) => None,
+                    (Some(prec), Some(scale)) => Some((prec, scale)),
+                    _ => None,
+                },
             )),
         ),
         "money" | "_money" => (Float, None),
-        "pg_lsn" | "_pg_lsn" => (Unsupported(full_data_type.into()), None),
-        "time" | "_time" => (DateTime, Some(PostgresType::Time(precision.time_precision()))),
-        "timetz" | "_timetz" => (
-            DateTime,
-            Some(PostgresType::TimeWithTimeZone(precision.time_precision())),
-        ),
-        "interval" | "_interval" => (Duration, Some(PostgresType::Interval(precision.time_precision()))),
-        "timestamp" | "_timestamp" => (DateTime, Some(PostgresType::Timestamp(precision.time_precision()))),
+        "pg_lsn" | "_pg_lsn" => unsupported(),
+        "time" | "_time" => (DateTime, Some(PostgresType::Time(precision.time_precision))),
+        "timetz" | "_timetz" => (DateTime, Some(PostgresType::TimeWithTimeZone(precision.time_precision))),
+        "interval" | "_interval" => (Duration, Some(PostgresType::Interval(precision.time_precision))),
+        "timestamp" | "_timestamp" => (DateTime, Some(PostgresType::Timestamp(precision.time_precision))),
         "timestamptz" | "_timestamptz" => (
             DateTime,
-            Some(PostgresType::TimestampWithTimeZone(precision.time_precision())),
+            Some(PostgresType::TimestampWithTimeZone(precision.time_precision)),
         ),
-        "tsquery" | "_tsquery" => (Unsupported(full_data_type.into()), None),
-        "tsvector" | "_tsvector" => (Unsupported(full_data_type.into()), None),
-        "txid_snapshot" | "_txid_snapshot" => (Unsupported(full_data_type.into()), None),
+        "tsquery" | "_tsquery" => unsupported(),
+        "tsvector" | "_tsvector" => unsupported(),
+        "txid_snapshot" | "_txid_snapshot" => unsupported(),
         "inet" | "_inet" => (String, None),
         //geometric
-        "box" | "_box" => (Unsupported(full_data_type.into()), None),
-        "circle" | "_circle" => (Unsupported(full_data_type.into()), None),
-        "line" | "_line" => (Unsupported(full_data_type.into()), None),
-        "lseg" | "_lseg" => (Unsupported(full_data_type.into()), None),
-        "path" | "_path" => (Unsupported(full_data_type.into()), None),
-        "polygon" | "_polygon" => (Unsupported(full_data_type.into()), None),
-        _ => (Unsupported(full_data_type.into()), None),
+        "box" | "_box" => unsupported(),
```

Comment:
Not to this PR, but maybe we should have user-facing errors for unsupported types?

Reply:
this does not error. its a closure that instantiates the actual type Unsupported ;-)

Reply:
Probably easy to misinterpret, I'll change the name.